lib/data/external/PfsClient.js

```diff
@@ -0,0 +1,148 @@
+const arsenal = require('arsenal');
+const errors = arsenal.errors;
+const werelogs = require('werelogs');
+const createLogger = require('../multipleBackendLogger');
+const { logHelper } = require('./utils');
+
+class PfsClient {
+    constructor(config) {
+        const { host, port } = config.endpoint;
+
+        this.clientType = 'pfs';
+        this.type = 'PFS';
+        this._bucketName = config.bucketName;
+        this._bucketMatch = config.bucketMatch;
+        this._serverSideEncryption = config.serverSideEncryption;
+        this._dataStoreName = config.dataStoreName;
+        this._supportsVersioning = config.supportsVersioning;
+        this._mountPath = config.mountPath;
+        this._logger = new werelogs.Logger('PfsClient');
+        this._restClient = new arsenal.network.rest.RESTClient({
+            host,
+            port,
+            isPassthrough: true,
+        });
+    }
+
+    setup(cb) {
+        return cb();
+    }
+
+    _createFsKey(requestBucketName, requestObjectKey, bucketMatch) {
+        if (bucketMatch) {
+            return requestObjectKey;
+        }
+        return `${requestBucketName}/${requestObjectKey}`;
+    }
+
+    toObjectGetInfo(objectKey, bucketName) {
+        return {
+            key: this._createFsKey(bucketName, objectKey, this._bucketMatch),
+            dataStoreName: this._dataStoreName,
+        };
+    }
+
+    put(stream, size, keyContext, reqUids, callback) {
+        const log = createLogger(reqUids);
+        if (keyContext.metaHeaders['x-amz-meta-mdonly'] === 'true') {
+            const b64 = keyContext.metaHeaders['x-amz-meta-md5chksum'];
+            let md5 = null;
+            if (b64) {
+                md5 = new Buffer(b64, 'base64').toString('hex');
+            }
+            return callback(null, keyContext.objectKey, '',
+                keyContext.metaHeaders['x-amz-meta-size'],
+                md5
+            );
+        }
+        logHelper(log, 'error', 'Not implemented', errors.NotImplemented,
+            this._dataStoreName, this.clientType);
+        return callback(errors.NotImplemented);
+    }
+
+    get(objectGetInfo, range, reqUids, callback) {
+        const log = createLogger(reqUids);
+        this._restClient.get(objectGetInfo.key, range, reqUids, (err, rs) => {
+            if (err) {
+                logHelper(log, 'error', 'err from data backend', err,
+                    this._dataStoreName, this.clientType);
+                return callback(err, null);
+            }
+            return callback(null, rs);
+        });
+    }
+
+    delete(objectGetInfo, reqUids, callback) {
+        const log = createLogger(reqUids);
+        logHelper(log, 'error', 'Not implemented', errors.NotImplemented,
+                this._dataStoreName, this.clientType);
+        return callback(errors.NotImplemented);
+    }
+
+    healthcheck(location, callback) {
+        const fsResp = {};
```

Comment:
If this is not implemented yet, you can add a `TODO:`, it's easier to follow up.